framework/TestFiles.py

```diff
@@ -0,0 +1,244 @@
+    @ In, comment, string, a comment printed out if it fails
+    @ In, res, bool, the tested value
+    @ Out, res, bool, True if test
+  """
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking bool",comment,'|',res,'is not True!')
+      results["fail"] += 1
+  return res
+
+def checkSame(comment,value,expected,update=True):
+  """
+    This method is aimed to compare two identical things
+    @ In, comment, string, a comment printed out if it fails
+    @ In, value, float, the value to compare
+    @ In, expected, float, the expected value
+    @ Out, res, bool, True if same
+  """
+  res = value == expected
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking string",comment,'|',value,"!=",expected)
+      results["fail"] += 1
+  return res
+
+def checkArray(comment,first,second,dtype,tol=1e-10,update=True):
+  """
+    This method is aimed to compare two arrays
+    @ In, comment, string, a comment printed out if it fails
+    @ In, value, float, the value to compare
+    @ In, expected, float, the expected value
+    @ In, tol, float, optional, the tolerance
+    @ Out, res, bool, True if same
+  """
+  res = True
+  if len(first) != len(second):
+    res = False
+    print("checking answer",comment,'|','lengths do not match:',len(first),len(second))
+  else:
+    for i in range(len(first)):
+      if dtype == float:
+        pres = checkFloat('',first[i],second[i],tol,update=False)
+      elif dtype in (str,unicode):
+        pres = checkSame('',first[i],second[i],update=False)
+      if not pres:
+        print('checking array',comment,'|','entry "{}" does not match: {} != {}'.format(i,first[i],second[i]))
+        res = False
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      results["fail"] += 1
+  return res
+
+def checkNone(comment,entry,update=True):
+  """
+    Checks if entry is None.
+    @ In, comment, string, a comment printed out if it fails
+    @ In, entry, object, entity to evaluate for None
+    @ In, update, bool, optional, if True then will update results
+    @ Out, res, bool, True if entry is None
+  """
+  res = entry is None
+  if update:
+    if res:
+      results["pass"] += 1
+    else:
+      print("checking answer",comment,'|','"{}" is not None!'.format(entry))
+      results["fail"] += 1
+  return res
+
+def checkFails(comment,errstr,function,update=True,args=None,kwargs=None):
+  """
+    Checks if entry is None.
+    @ In, comment, string, a comment printed out if it fails
+    @ In, errstr, str, message given by error to check against
+    @ In, function, method, function to evaluate that is expected to fail
+    @ In, update, bool, optional, if True then will update results
+    @ In, args, list, additional arguments to pass to function
+    @ In, kwargs, dict, additional keyword arguments to pass to function
+    @ Out, res, bool, True if entry is None
+  """
```

Comment:
Please update the comments description and the output descriptions

Reply:
done

Reply:
good to me.